src/cal/mod.rs

```diff
@@ -43,13 +142,20 @@ pub struct Event {
     pub duration: Duration,
 }
 
-#[allow(dead_code)]
 impl Event {
-    pub fn overlap(&self, other: &Event) -> bool {
-        match self.date.cmp(&other.date) {
-            Ordering::Less => self.date + self.duration > other.date,
-            Ordering::Greater => other.date + other.duration > self.date,
-            Ordering::Equal => true,
+    pub fn overlap(&self, other: &Event) -> Option<Event> {
+        let start = max(self.date, other.date);
```

Comment:
I don't think this should return an Event, it should return the interval of time over which they overlap.

Implementation-wise, probably simpler to order the two events, then get the end datetime of the left one and the start datetime of the right one. e.g. in pseudocode

``` Rust
let left = min(self, other);
let right = max(self, other);

let end = left.date + left.duration;
let start = right.date;

if start <= end {
    Some(/* blah */)
} else {
    None
} 
```

Though you'll probably implement the interval intersection code separately, then this simply reduces to 

``` Rust
Interval::new(self.date, self.date + self.duration).intersect(other.date, other.date + other.duration)
```

or however you implement Interval.

Reply:
I'll implement interval separately and make a new pull request.
Without having intervals you pretty much have to return an event.
I'll change the implementation.